src/modules/hyprland/workspaces.cpp

```diff
@@ -284,10 +284,10 @@ Workspace::Workspace(const Json::Value &workspace_data)
       windows_(workspace_data["windows"].asInt()),
       active_(true),
       is_special_(false) {
-  if (name_.find("name:") == 0) {
+  if (name_.starts_with("name:")) {
     name_ = name_.substr(5);
-  } else if (name_.find("special") == 0) {
-    name_ = id_ == -99 ? name_ : name_.substr(13);
```

Comment:
@MightyPlaza why was this substr(13) and not 8?

Reply:
yes it should (maybe I was thinking "special:name:<name_here>" but that isn't what happens)

[special.txt](https://github.com/Alexays/Waybar/files/12177873/special.txt)
this diff fixes it and issues I found in the the on_click function (no idea how I didn't notice this before)